prov/rxm/src/rxm_cq.c

```diff
@@ -574,8 +574,12 @@ static ssize_t rxm_cq_handle_comp(struct rxm_ep *rxm_ep,
 			rxm_ep_msg_mr_closev(tx_entry->mr, tx_entry->count);
 		return rxm_finish_send_nobuf(tx_entry);
 	case RXM_TX:
-		assert(comp->flags & (FI_SEND | FI_WRITE));
+		assert(comp->flags & (FI_SEND));
 		return rxm_finish_send(tx_entry);
+	case RXM_TX_RMA:
+		assert(comp->flags & (FI_SEND));
```

Comment:
Actually, for FI_READ the tx_entry state would be RXM_TX_NOBUF and not RXM_TX_RMA. That's what the old code did. I think the new code does the same as well.

Reply:
Sorry, I forgot to provide some clarifications -- for FI_READ the `tx_entry::state` can be `RXM_TX_RMA` for only postponed case.  It uses iovec array (field of rxm_rma_buf) and stores user's iovec there.

Reply:
I'm looking at the first commit by itself,  Based on that, this removes FI_WRITE from RXM_TX, so I would expect the new case to use FI_WRITE rather than FI_SEND, which agrees with Arun's initial comment.